manager/controllers/systems_advisories_view.go

```diff
@@ -32,13 +32,28 @@ type systemsAdvisoriesDBLoad struct {
 var systemsAdvisoriesSelect = database.MustGetSelect(&systemsAdvisoriesDBLoad{})
 
 func systemsAdvisoriesQuery(acc int, systems []SystemID, advisories []AdvisoryName) *gorm.DB {
-	query := database.SystemAdvisories(database.Db, acc).
-		Select(systemsAdvisoriesSelect).
-		Joins("join advisory_metadata am on am.id = sa.advisory_id").
-		Where("sp.inventory_id::text in (?)", systems).
-		Where("am.name in (?)", advisories).
-		Order("sp.inventory_id, am.id")
-
+	var query *gorm.DB
+	switch {
+	case len(systems) == 0:
+		query = database.SystemAdvisories(database.Db, acc).
+			Select(systemsAdvisoriesSelect).
+			Joins("join advisory_metadata am on am.id = sa.advisory_id").
+			Where("am.name in (?)", advisories).
+			Order("sp.inventory_id, am.id")
+	case len(advisories) == 0:
+		query = database.SystemAdvisories(database.Db, acc).
+			Select(systemsAdvisoriesSelect).
+			Joins("join advisory_metadata am on am.id = sa.advisory_id").
+			Where("sp.inventory_id::text in (?)", systems).
+			Order("sp.inventory_id, am.id")
+	default:
+		query = database.SystemAdvisories(database.Db, acc).
+			Select(systemsAdvisoriesSelect).
+			Joins("join advisory_metadata am on am.id = sa.advisory_id").
+			Where("sp.inventory_id::text in (?)", systems).
+			Where("am.name in (?)", advisories).
+			Order("sp.inventory_id, am.id")
+	}
```

Comment:
I don't like having the nearly-the-same query 3 times ... what about building it incrementally
```
query = database.SystemAdvisories(database.Db, acc).
			Select(systemsAdvisoriesSelect).
			Joins("join advisory_metadata am on am.id = sa.advisory_id").
			Order("sp.inventory_id, am.id")
if len(advisories) > 0 {
	query = query.Where("am.name in (?)", advisories)
}
if len(systems) > 0 {
	query = query.Where("sp.inventory_id::text in (?)", systems)
}
```

Reply:
sure, I' still not familiar with gorm and I didn't know it works this way 😄